context/app/api/vitessce.py

```diff
@@ -285,8 +302,13 @@ def _build_vitessce_conf(self):
             found_tiles = _get_matches(file_paths_found, TILE_REGEX)
             confs = []
             for tile in sorted(found_tiles):
-                new_conf = copy.deepcopy(conf)
-                layers = [self._build_layer_conf(file, tile) for file in files]
+                # If there are no cell segmentations, remove the non-imaging parts of the assay.
+                if f"{CODDEX_SPRM_PATH}/{tile}.cell-sets.json" not in file_paths_found:
+                    new_conf = copy.deepcopy(TILED_SPRM_IMAGING_ONLY)
+                    layers = [self._build_layer_conf(files[0], tile)]
+                else:
+                    new_conf = copy.deepcopy(conf)
+                    layers = [self._build_layer_conf(file, tile) for file in files]
```

Comment:
```suggestion
                if f"{CODDEX_SPRM_PATH}/{tile}.cell-sets.json" not in file_paths_found:
                    new_conf = copy.deepcopy(TILED_SPRM_IMAGING_ONLY)
                    files_for_layers = [files[0]]
                else:
                    new_conf = copy.deepcopy(conf)
                    files_for_layers = files
                layers = [self._build_layer_conf(file, tile) for file in files_for_layers]
```
slight preference to pull this out...